doc: updating Ninja guide (#32629)
nodejs/node: doc/guides/building-node-with-ninja.md

doing so can be significantly quicker than using `make`. Please see
[Ninja's site][Ninja] for installation instructions (Unix only).

> [Ninja][] is already supported at the Makefile, the only thing you need to
> execute is `./configure --ninja`, and this will configure the project to run
> the regular `make` commands with [Ninja][].

Reviewer: ```suggestion
[Ninja][] is supported in the Makefile. Run `./configure --ninja` to configure
the project to run the regular `make` commands with [Ninja][].
```

> For example, running `make` will execute `ninja -C out/Release` internally
> to produce a compiled release binary, and will also execute
> `ln -fs out/Release/node node`, so that you can execute `./node` at
> the project's root.

Reviewer: ```suggestion
For example, `make` will execute `ninja -C out/Release` internally
to produce a compiled release binary. It will also execute
`ln -fs out/Release/node node` so that you can execute `./node` at
the project's root.
```

`[finished/total]` build steps. This is useful output that `make` does not
produce and is one of the benefits of using Ninja. Also, Ninja will likely
compile much faster than even `make -j4` (or

> `-j<number of processor threads on your machine>`), you can still pass the
> number of processes to run for [Ninja][] using the environment variable `JOBS`,
> and this will be the equivalent to the `-j` parameter in the regular `make`:

Reviewer: ```suggestion
`-j<number of processor threads on your machine>`). You can still pass the
number of processes to run for Ninja using the environment variable `JOBS`.
This will be the equivalent to the `-j` parameter in the regular `make`:
```

```bash
$ JOBS=12 make
```

## Producing a debug build

> Is possible to create an alias to produce a debug build, rather than a

Reviewer 2: Is -> It's ?

Author: Thanks for reviewing, just addressed your feedback

release build as shown below:

> `alias nnodedebug='./configure --ninja && ninja -C out/Debug && ln -fs

Reviewer 3: `./configure --ninja --debug` + `make` also works

Author: Thanks for your suggestion, changed the instructions to match your suggestion

> It is possible to create a debug build, rather than a release build as shown
> below:

Reviewer: ```suggestion
To create a debug build rather than a release build:
```

out/Debug/node node_g'`

[Ninja]: https://ninja-build.org/

> [this commit]: https://github.com/nodejs/node/pull/27504/commits/46eb532a2a144ebd0fe1aa91a2b4a9e9372aa902

Reviewer 3: I don't think it's common for us to link which commits introduced certain build options. Not a blocker, but we should probably remove it.

Author: Got it, removed the reference too :)
